File: CSharp/leetcodesln/ContainerWithMostWater.cs
using System;

namespace leetcodesln
{
    public class ContainerWithMostWater
    {
        public int MaxArea(int[] height)
        {
            int l = 0, r = height.Length - 1, max = 0;

            while (l < r)
            {
                max = Math.Max(max, Math.Min(height[l], height[r]) * (r - l));
                if (height[l] > height[r]) --r;
                else ++l;
            }
            return max;
        }
    }
}

File: Go/questions/ContainerWithMostWater.go
package questions

import . "github.com/leetcode/common"

func MaxArea(height []int) int {
	ans, l, r := 0, 0, len(height)-1

	for l < r {
		cur := (r - l) * Min(height[l], height[r])
		ans = Max(ans, cur)
		if height[l] < height[r] {
			l++
		} else {
			r--
		}
	}
	return ans
}
